### ISSUE TITLE
The Triangle class

### ISSUE THREAD

**Issue by OrtnerMichael** [magpylib/magpylib#568]
Hi everyone, but specifically @Alexboiboi @fslanovc  and @peter-leitner 

After a lot of thinking and discussion with you I propose the following interface layout for the Facet code: We introduce one core function `magnet_facet_field` and three classes for the object oriented interface: `magnet.Tetrahedron` , `magnet.FacetBody`, and `misc.Facet`

The Facet class
-------------------
Parameters: `magnetization` (the usual) and `vertices` shape (3,3) or (n,3,3)

This class contains only the bare triangular facet code and is for users who like to customize, e.g. work only with single faces, or non-closed meshes because they know that parallel faces to the magnetization do not contribute.

Users have to think by themselves about inside/outside, and about the face orientations.

Facet objects are not magnets (not even if all faces are properly arranged, because there is no inside-outside logic), so the `Facet` goes to the `misc` sub-package.

This already works in the [facet_MO](https://github.com/magpylib/magpylib/tree/facet_MO) branch 

The Tetrahedron class
-------------------------
Parameters: `magnetization` (the usual) and `vertices` shape (4,3)

This class is introduced because the Tetrahedron is a very basic shape which can be used to construct a lot of complex forms.

The Tetrahedron is very user-friendly: The faces are automatically arranged in the computation and inside-outside is computed. It behaves like a magnet, so it goes to the `magnet` sub-package.

This already works in the [facet_MO](https://github.com/magpylib/magpylib/tree/facet_MO) branch

**Comment by Alexboiboi** [comment]
Should we also introducing a style option to display each facet orientation?
I'd suggest to use a small cone with the base in the plane of the facet. I would not use an arrow in order not to confuse it with the magnetization arrow or with the dipole object. Another options would be to colorize each side of each facet differently.

@OrtnerMichael  what do you tink?

**Comment by Alexboiboi** [comment]
> Should we also introducing a style option to display each facet orientation? I'd suggest to use a small cone with the base in the plane of the facet. I would not use an arrow in order not to confuse it with the magnetization arrow or with the dipole object. Another options would be to colorize each side of each facet differently.
> 
> @OrtnerMichael what do you tink?

@OrtnerMichael, would be nice to answer this ;)

**Comment by Alexboiboi** [comment]
My thoughts on this topic:

## Naming
In general, I would prefer more explicit naming. The currently named `Facet` object is a particular object that does not represent a real physical object, so I agree that in should be in the `misc` category. So far in the library, all sources are named after their unambiguous shape. Since the `Facet` object only allows triangles I'd rename it to `Triangle`. Later if we really want we can still implement a `Facet` class which allows for arbitrary polygon shapes.

## Functionality
I find it a bit odd, that you can allow multiple disconnected triangles inside a single `Facet` object. I would restrict the `Facet` class to build a single triangle and would reserve the grouping of multiple triangles for the future `TriangularMesh` **magnet** class, where it makes more sense to consider it as a single object. It anyway does not change the internal vectorized way of calculating the magnetic field. We should just check the potential overhead of creating many `Facet`/`Triangle` instances to build an object, but if you have to, it may make more sense just to use the `TriangularMesh` class (which should allow to disable the mesh checks and allow open or disconnected bodies).

## Documentation
In my opinion, the current example in the docs already impinges too much on the `TriangularMesh` functionality. There should be shown an use case where `Facet`/`Triangle` is prefered over a `TriangularMesh` (e.g. prism with triangular base axially magnetized represented by two triangles, since side walls do not contribute to the magnetic field in this case). Otherwise I don't really see the benefit of implementing this class in the first place. Here I cannot help much, @OrtnerMichael you must tell your perspective as a physicist ;) 

I may update this comment, if more thoughts come to mind.

**Comment by OrtnerMichael** [comment]
@Alexboiboi Sorry for my late answering - it was a challenging week - keep bugging me if I don't answer! 

I implemented practically all changes you suggested in #574 . I agree that it makes this much better, more fundamental, simpler and just generally nicer. I have already added doc examples and worked a bit with the new Triangle class - it feels good :). I have removed the old examples, also the Pyvista example - maybe we leave that for the `TriMesh` (or `TriangularMesh` ?).

I am in strong favor of your idea to graphically show the orientation of each Triangle object - this would be extremely helpful. The cone sounds good to me ! This should add another graphic style attribute as well that can be turned off with `style_orientation_show=False`.



Also for the
